src/Nethermind/Nethermind.Core/Caching/ICache.cs

```diff
@@ -14,13 +14,17 @@
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
 
+using Nethermind.Core.Crypto;
+
 namespace Nethermind.Core.Caching
 {
     public interface ICache<in TKey, TValue>
     {
         void Clear();
         TValue Get(TKey key);
+        bool TryGet(TKey key, out TValue value);
```

Comment:
something smells here - we currently do not differentiate null and lack of value so this interface implementations will not behave properly? (or maybe?)

Reply:
It will behave properly as we can query Dictionary<> underneath

Reply:
ok